python/lsst/analysis/tools/actions/plot/scatterplotWithTwoHists.py

```diff
@@ -173,45 +179,41 @@ class ScatterPlotWithTwoHists(PlotAction):
+            color="firebrick",
+        ),
+        "unknown": DataTypeDefaults(
+            suffix_stat="Unknown",
+            suffix_xy="Unknown",
+            color="green",
+        ),
+        "any": DataTypeDefaults(
+            suffix_stat="Any",
+            suffix_xy="",
+            color="purple",
+        ),
+    }
 
     def getInputSchema(self) -> KeyedDataSchema:
         base: list[tuple[str, type[Vector] | ScalarType]] = []
-        if "stars" in self.plotTypes:  # type: ignore
-            base.append(("xStars", Vector))
-            base.append(("yStars", Vector))
-            base.append(("starsHighSNMask", Vector))
-            base.append(("starsLowSNMask", Vector))
-            # statistics
-            for name in self._stats:
-                base.append((f"{{band}}_highSNStars_{name}", Scalar))
-                base.append((f"{{band}}_lowSNStars_{name}", Scalar))
-        if "galaxies" in self.plotTypes:  # type: ignore
-            base.append(("xGalaxies", Vector))
-            base.append(("yGalaxies", Vector))
-            base.append(("galaxiesHighSNMask", Vector))
-            base.append(("galaxiesLowSNMask", Vector))
-            # statistics
-            for name in self._stats:
-                base.append((f"{{band}}_highSNGalaxies_{name}", Scalar))
-                base.append((f"{{band}}_lowSNGalaxies_{name}", Scalar))
-        if "unknown" in self.plotTypes:  # type: ignore
-            base.append(("xUnknown", Vector))
-            base.append(("yUnknown", Vector))
-            base.append(("unknownHighSNMask", Vector))
-            base.append(("unknownLowSNMask", Vector))
-            # statistics
-            for name in self._stats:
-                base.append((f"{{band}}_highSNUnknown_{name}", Scalar))
-                base.append((f"{{band}}_lowSNUnknown_{name}", Scalar))
-        if "any" in self.plotTypes:  # type: ignore
-            base.append(("x", Vector))
-            base.append(("y", Vector))
-            base.append(("anyHighSNMask", Vector))
-            base.append(("anySNMask", Vector))
-            # statistics
-            for name in self._stats:
-                base.append((f"{{band}}_highSNAny_{name}", Scalar))
-                base.append((f"{{band}}_lowSNAny_{name}", Scalar))
+        for datatype, defaults in self._datatypes.items():
+            if datatype in self.plotTypes:  # type: ignore
+                base.append((f"x{defaults.suffix_xy}", Vector))
+                base.append((f"y{defaults.suffix_xy}", Vector))
+                base.append((f"{datatype}HighSNMask", Vector))
+                base.append((f"{datatype}LowSNMask", Vector))
+                # statistics
+                for name in self._stats:
+                    base.append((f"{{band}}_highSN{defaults.suffix_stat}_{name}", Scalar))
+                    base.append((f"{{band}}_lowSN{defaults.suffix_stat}_{name}", Scalar))
         base.append(("lowSnThreshold", Scalar))
```

Comment:
This is great

Reply:
Isn't it? So much so that I added similar loop roll-ups for the marginal histograms. Also, it now loops over self.plotTypes rather than looping over all possible plot types and checking if they're in self.plotTypes, so you can set the order of plotting however you like.